### PR TITLE
Richard Severich Bank OCR

### PR DESCRIPTION
**Pull request by RichardSeverich**

### CODE REVIEW

File: src/main/java/org/fundacionjala/coding/richardseverich/BankOCR.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            resultNumber = (band) ? resultNumber.append("?") : resultNumber.append("");
+        }
+        return resultNumber.toString();
+    }
+
+    /**
+     * @param num Is the number already converted.
+     * @return Returns the number with Err, ILL.
+     */
+    public String lineOutPut(String num) {
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/coding/richardseverich/BankOCR.java]: Rename this method.
use one more descriptive.

File: src/main/java/org/fundacionjala/coding/richardseverich/BankOCR.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        StringBuilder resultNumber = new StringBuilder();
+        resultNumber.append(num);
+        if (num.contains("?")) {
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/coding/richardseverich/BankOCR.java]: Please use a constant for "?"

File: src/main/java/org/fundacionjala/coding/richardseverich/BankOCR.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        for (int i = 0; i < NINE; i++) {
+            numDynamic[i] = num[i].append(entryLine.substring(j, j + THREE)).toString();
+            j += THREE;
+        }
+    }
+
+    /**
+     *
+     */
+    public void initializeNumberDynamicBuilder() {
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/coding/richardseverich/BankOCR.java]: Try to remove this method in your solution.

Comment (carledriss) [reply src/main/java/org/fundacionjala/coding/richardseverich/BankOCR.java]: Consider using a List instead of array.

Comment (RichardSeverich) [reply src/main/java/org/fundacionjala/coding/richardseverich/BankOCR.java]: The method was deleted.

File: src/main/java/org/fundacionjala/coding/richardseverich/BankOCR.java (hunk 1 above)
Comment (RichardSeverich) [reply src/main/java/org/fundacionjala/coding/richardseverich/BankOCR.java]: The method was renamed

File: src/main/java/org/fundacionjala/coding/richardseverich/BankOCR.java (hunk 2 above)
Comment (RichardSeverich) [reply src/main/java/org/fundacionjala/coding/richardseverich/BankOCR.java]: Now I'm using a constant for "?". Thanks for the Feedback.

File: src/main/java/org/fundacionjala/coding/richardseverich/BankOCR.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+     */
+    public String identifyErrOrIll(String num) {
+        StringBuilder resultNumber = new StringBuilder();
+        resultNumber.append(num);
+        if (num.contains(QUESTION_MARK)) {
+            resultNumber.append(" ILL");
+        } else if (!isValidChecksum(num)) {
+            resultNumber.append(" ERR");
+        }
+        return resultNumber.toString();
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/coding/richardseverich/BankOCR.java:68]: Instead of using a StringBuilder you can use a String.format

**Comment by carledriss** [comment]
LGTM !

**Pull request merged**